example_plugins/baz/src/BazService.ts

```diff
@@ -7,39 +7,63 @@ import {
   toPromise
 } from '@elastic/kbn-observable';
 
-import {
-  ElasticsearchService,
-  KibanaConfig,
-  KibanaRequest
-} from '@elastic/kbn-types';
+import { ElasticsearchService, KibanaConfig } from '@elastic/kbn-types';
 
 export class BazService {
+  private readonly _esService: any; // help, this is really a function
+
+  // NB: When we finally create BazService, we have a request
+  // object. That's why we can pass in headers and get an elasticsearch
+  // service scoped to the request.
+  //
+  // But we can also choose the cluster to bind all our elasticsearch
+  // calls to. What if we could do both in one step at this high level?
   constructor(
-    private readonly req: KibanaRequest,
+    private readonly headers: string,
     private readonly kibanaConfig$: Observable<KibanaConfig>,
     private readonly elasticsearchService: ElasticsearchService
-  ) {}
+  ) {
+    // NB: This is where we should bind to the right cluster for
+    // elasticsearch service.
+    //
+    // We could do it directly in the exposed BazService methods,
+    // but that feels too late, and it also feels like a lot of
+    // extra cruft for plugin developers.
+    //
+    // We can assume all the routes for a service will call internal
+    // services (like elasticsearch) with the same settings.
+    // If Baz Plugin needs to hit a different cluster, then there
+    // can be a BazBarService, sister to BazService, that binds to
+    // that other cluster.
+    //
+    // The drawback might be: is this too early to bind, since
+    // the configuration could change? We're creating this service
+    // and then immediately handling the request, so I think it's fine,
+    // but there's nothing enforcing plugin developers to write
+    // this way.
+
+    this._esService = this.elasticsearchService.getScopedCluster(
```

Comment:
I agree that we can bind now. The only suggestion I have is maybe `bazService` should just receive a `Cluster` instance? That way we don't have to inject headers down here. So we'd do `elasticsearchService.getScopedCluster` in the route handler instead, and inject the `Cluster`? What do you think?

Reply:
🤔 :+1: